vault/seal/ocikms/ocikms.go

```diff
@@ -268,29 +284,71 @@ func (k *OCIKMSSeal) Decrypt(ctx context.Context, in *physical.EncryptedBlobInfo
 		return nil, fmt.Errorf("given input for decryption is nil")
 	}
 
+	// Default to mechanism used before key info was stored
+	if in.KeyInfo == nil {
+		in.KeyInfo = &physical.SealKeyInfo{
+			Mechanism: OCIKMSEncrypt,
+		}
+	}
+
 	requestMetadata := k.getRequestMetadata()
+	var plaintext []byte
+	switch in.KeyInfo.Mechanism {
+	case OCIKMSEncrypt:
+		cipherText := string(in.Ciphertext)
+		decryptedDataDetails := keymanagement.DecryptDataDetails{
+			KeyId:      &k.keyID,
+			Ciphertext: &cipherText,
+		}
+		input := keymanagement.DecryptRequest{
+			DecryptDataDetails: decryptedDataDetails,
+			RequestMetadata:    requestMetadata,
+		}
 
-	cipherText := string(in.Ciphertext)
-	decryptedDataDetails := keymanagement.DecryptDataDetails{
-		KeyId:      &k.keyID,
-		Ciphertext: &cipherText,
-	}
-	input := keymanagement.DecryptRequest{
-		DecryptDataDetails: decryptedDataDetails,
-		RequestMetadata:    requestMetadata,
-	}
+		output, err := k.cryptoClient.Decrypt(ctx, input)
+		if err != nil {
+			metrics.IncrCounter(metricDecryptFailed, 1)
+			return nil, errwrap.Wrapf("error decrypting data: {{err}}", err)
+		}
+		plaintext, err = base64.StdEncoding.DecodeString(*output.Plaintext)
+		if err != nil {
+			metrics.IncrCounter(metricDecryptFailed, 1)
+			return nil, errwrap.Wrapf("error base64 decrypting data: {{err}}", err)
+		}
+		k.logger.Debug("successfully decrypted")
+	case OCIKMSEnvelopeAESGCMEncrypt:
+		cipherTextBlob := string(in.KeyInfo.WrappedKey)
+		decryptedDataDetails := keymanagement.DecryptDataDetails{
+			KeyId:      &k.keyID,
+			Ciphertext: &cipherTextBlob,
+		}
+		input := keymanagement.DecryptRequest{
+			DecryptDataDetails: decryptedDataDetails,
+			RequestMetadata:    requestMetadata,
+		}
+		output, err := k.cryptoClient.Decrypt(ctx, input)
+		if err != nil {
+			metrics.IncrCounter(metricDecryptFailed, 1)
+			return nil, errwrap.Wrapf("error decrypting data: {{err}}", err)
+		}
+		plaintext, err = base64.StdEncoding.DecodeString(*output.Plaintext)
```

Comment:
`decryptedKey` (or something) would be better, since `plaintext` is used later to me something else.